### DISCUSSION TITLE
Use GLSPActionDispatcher in PropertyView

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#912] [Q&A] [answered]
Hi,
    I create a property view extension to show glsp node property, Now when I change the value in property view I want to update to glspserver model data. I create a custom operation, you know I need GLSPActionDispatcher to dispatch my operation to server. But When I inject the GLSPActionDispatcher I get error. It seem that GLSPActionDispatcher can not direct inject to another extension module. is there any method can I dispatch my operation to server?
![image](https://user-images.githubusercontent.com/4537809/217770106-ce651df4-c1d9-43f3-9840-eff40f58199c.png)
![image](https://user-images.githubusercontent.com/4537809/217771473-7e7af53e-34c8-4102-80ea-96f6cc7ab30f.png)

**Comment by ndoschek** [comment, marked as answer]
Hi @wss29,

This topic just recently came up, may I redirect you to this discussion thread: https://github.com/eclipse-glsp/glsp/discussions/888
If you have any further questions, I'm glad to answer them.

HTH and best regards,
Nina

**Reply by wss29** [reply]
Hi,
    I find when I edit property value in proertyview, this.shell.activeWidget || this.shell.currentWidget is PropertyViewWidget. I can not get GLSPDiagramWidget, in this case I can not dispatch operation.
![image](https://user-images.githubusercontent.com/4537809/217974458-ba2578ac-7820-42b6-93b3-bac21610574f.png)

**Reply by ndoschek** [reply]
Hi @wss29,
in that case you could fetch the diagramWidget via a separate helper function, for example like this:

```javascript
protected getDiagramWidget(diagramUri: string): GLSPDiagramWidget | undefined {
   for (const widget of this.shell.widgets) {
       if (widget instanceof GLSPDiagramWidget && widget.uri.toString() === diagramUri) {
           return widget;
       }
    }
    return undefined;
}
```
The current model URI can be stored in the `widget-provider` and be used to fetch the diagram widget.

HTH and best regards,
Nina
